pennylane/vqe/vqe.py

```diff
@@ -490,30 +490,12 @@ def __init__(
         """Bool: Records if multiple devices are input"""
 
         tape_mode = qml.tape_mode_active()
-        if tape_mode:
-
-            d = device[0] if self._multiple_devices else device
-            w = d.wires.tolist()
-
-            try:
-                qml.disable_tape()
-
-                @qml.qnode(d, interface=interface, diff_method=diff_method, **kwargs)
-                def qnode_for_metric_tensor_in_tape_mode(*qnode_args, **qnode_kwargs):
-                    """The metric tensor cannot currently be calculated in tape-mode QNodes. As a
-                    short-term fix for ExpvalCost, we create a non-tape mode QNode just for
-                    calculation of the metric tensor. In doing so, we reintroduce the same
-                    restrictions of the old QNode but allow users to access new functionality
-                    such as measurement grouping and batch execution of the gradient."""
-                    ansatz(*qnode_args, wires=w, **qnode_kwargs)
-                    return qml.expval(qml.PauliZ(0))
-
-                self._qnode_for_metric_tensor_in_tape_mode = qnode_for_metric_tensor_in_tape_mode
-            finally:
-                qml.enable_tape()
-
         self._optimize = optimize
 
+        self.qnodes = qml.map(
+            ansatz, observables, device, interface=interface, diff_method=diff_method, **kwargs
+        )
```

Comment:
My only worry of moving this here is when `self._optimize is True`, then `self.qnodes` will be visible to the user but disconnected from the actual QNode used to evaluate.